pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go

```diff
@@ -189,12 +195,7 @@ func (r *managedReconcile) cleanUpAppliedManifestWorks(ctx context.Context, klus
 		}
 
 		// remove finalizer if exists
-		if mutated := removeFinalizer(&appliedManifestWorks.Items[index], appliedManifestWorkFinalizer); !mutated {
-			continue
-		}
-
-		_, err := r.managedClusterClients.appliedManifestWorkClient.Update(ctx, &appliedManifestWorks.Items[index], metav1.UpdateOptions{})
-		if err != nil && !errors.IsNotFound(err) {
+		if err := patcher.RemoveFinalizer(ctx, &appliedManifestWorks.Items[index], appliedManifestWorkFinalizer); err != nil && !errors.IsNotFound(err) {
```

Comment:
when will we get a "IsNotFound" error?

Reply:
when the resource has been deleted.

Reply:
do we need to check this error in the common patcher package?

Reply:
yeah good point